source/_posts/react-firebase.md

```diff
@@ -0,0 +1,954 @@
+---
+title: Build a To-Do Application with React and Firebase
+contributor: Deji Adesoga
+avatar: deji-adesoga.png
+description: Learn how to use build a CRUD application by using React and Firebase 9.
+date: 2023-01-12
+tags: [react]
+id: react-firebase
+relatedPost: react-localstorage
+---
+
+
+# Build a To-Do Application with React and Firebase
+
+To-do applications are one of the ways you can use to manage a set of tasks. As developers, learning how to build a to-do application will also help to understand certain concepts one of which includes an understanding of how to build a CRUD(Create, Read, Update and Delete) application.
+
+In this article, you will learn how to build a to-do web app by making use of React.js and Firebase Database.
+
+## **Table of Contents**
+
+- [Prerequisites](#prerequisites)
+- [How to Create the Firebase Project](#how-to-setup-the-firebase-project)
+- [Creating the Firestore Database](#creating-the-firestore-database)
+- [How to Create the React Project](#how-to-create-the-react-project)
+- [Setting up the Project Structure](#setting-up-the-project-structure)
+- [How to Integrate Firebase in React](#how-to-integrate-firebase-in-react)
+- [Integrate Bootstrap 5 into React](#integrate-bootstrap-5-into-react)
+- [Designing the User Interface](#designing-the-user-interface)
+- [Adding Data to the Firestore in React](#adding-data-to-the-firestore-in-react)
+- [Fetching Data from the Firestore in React](#fetching-data-from-the-firestore-in-react)
+- [Deleting Data from the Firestore in React](#deleting-data-from-the-firestore-in-react)
+- [Updating Data in the Firestore in React](#updating-data-in-the-firestore-in-react)
+- [How to Integrate the Checkbox Functionality](#how-to-integrate-the-checkbox-functionality)
+- [How to order Data by Timestamp in Firebase](#how-to-order-data-by-timestamp-in-firebase)
+- [Conclusion](#conclusion)

```

Comment:
Please use `<!-- toc -->` instead of the contents. Examples are in the other blog posts.